wagtail/tests/test_management_commands.py

```diff
@@ -620,21 +620,26 @@ class TestPurgeRevisionsCommandForPages(TestCase):
     base_options = {}
 
     def setUp(self):
-        self.object = self.get_object()
-
-    def get_object(self):
-        # Find root page
+        # Create a page and its revision
         self.root_page = Page.objects.get(id=2)
-        self.page = SimplePage(
+        self.page = Page(
             title="Hello world!",
             slug="hello-world",
             content="hello",
             live=False,
         )
         self.root_page.add_child(instance=self.page)
         self.page.refresh_from_db()
-        return self.page
 
+        self.revision = Revision.objects.create(revision=1, page=self.page)
+        self.revision.user = self.user
+        self.revision.save()
+
+    def test_purge_revisions_protected_error(self):
+        # Run the purge_revisions command
+        with self.assertRaises(SystemExit):
+            self.call_command('purge_revisions', '--force')
```

Comment:
The changes in this test file do not seem to be related to the implementation changes at all. There is no `--force` option for the command...

Instead, we should have a model that references the `Revision` model with a `ForeignKey` that has `on_delete=models.PROTECT`, create an instance of that model with a stale revision pointed by the foreign key, and try to run the command.